TodoList/Sources/TodoList.Api/Controllers/ListItemsController.cs

```diff
@@ -1,78 +1,81 @@
-
-        public async Task<IHttpActionResult> GetAsync()
-        {
-            var items = await _repository.GetAllAsync(); 
-            return Ok(items);
-        }
-
-        public async Task<IHttpActionResult> GetAsync(Guid id)
-        {
-            var item = await _repository.GetAsync(id);
-            return Ok(item);
-        }
-
-        public async Task<IHttpActionResult> PostAsync(ListItem item)
-        {
-            ValidatePostItem(item);
-
-            if (!ModelState.IsValid)
-            {
-                return BadRequest(ModelState);
-            }
-            ListItem createdItem = await _repository.CreateAsync(item);
-            string location = _urlGenerator.GenerateUrl(createdItem);
-
-            return Created(location, createdItem);
-        }
-
-        public async Task<IHttpActionResult> PutAsync(ListItem item)
-        {
-            var updatedItem = await _repository.UpdateAsync(item);
-            return Ok(updatedItem);
-        }
-
-        public async Task<IHttpActionResult> DeleteAsync(Guid id)
-        {
-            var deletedItem = await _repository.DeleteAsync(id); 
-            return Ok(deletedItem);
-
-        }
-
-        private void ValidatePostItem(ListItem item)
-        {
-            if (item == null)
-            {
-                ModelState.AddModelError(String.Empty, "Item is null");
-                return;
-            }
-
-            if (string.IsNullOrWhiteSpace(item.Text))
-            {
-                ModelState.AddModelError(nameof(ListItem.Text), "Text is null or empty");
-            }
-
-            if (item.Id != Guid.Empty)
-            {
-                ModelState.AddModelError(nameof(ListItem.Id), "Guid is not empty");
-            }
-        }
-    }
+﻿using System;
+using System.Threading.Tasks;
+using System.Web.Http;
+using TodoList.Api.Services;
+using TodoList.Contracts.Models;
+using TodoList.Contracts.Repositories;
+
+namespace TodoList.Api.Controllers
+{
+    public class ListItemsController : ApiController
+    {
+        private readonly IListItemRepository _listItemsRepository;
+        private readonly IListItemUrlGenerator _urlGenerator;
+
+        public ListItemsController(IListItemRepository listItemsRepository, IListItemUrlGenerator generator)
+        {
+            _listItemsRepository = listItemsRepository;
+            _urlGenerator = generator;
+        }
+
+        public async Task<IHttpActionResult> GetAsync() 
+            => Ok(await _listItemsRepository.GetAllAsync());
+
+        public async Task<IHttpActionResult> GetAsync(Guid id)
+        {
+            ValidateItemId(id);
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            return Ok(await _listItemsRepository.GetAsync(id));
+        }
+
+        public async Task<IHttpActionResult> PostAsync(ListItem item)
+        {
+            ValidatePostItem(item);
+
+            if (!ModelState.IsValid)

```

Comment:
You might want to be a bit more consistent between methods. Here, I think the `if` is more related to the `ValidatePostItem` than to item creation, very similar to the case of `ValidateItemId` in `Get`. Thus the free line should rather be present between the `if`'s end and item creation.